projects/knora-ui/src/lib/viewer/values/color-value/color-value.component.spec.ts

```diff
@@ -0,0 +1,405 @@
+
+    });
+
+    it('should restore the initial displayed value', () => {
+
+      testHostComponent.mode = 'update';
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.colorValueComponent.mode).toEqual('update');
+
+      expect(testHostComponent.colorValueComponent.colorPickerComponent.readonly).toEqual(false);
+
+      expect(testHostComponent.colorValueComponent.form.valid).toBeFalsy();
+
+      // simulate user input
+      const newColor = '#g7g7g7';
+
+      testHostComponent.colorValueComponent.colorPickerComponent.value = newColor;
+      testHostComponent.colorValueComponent.colorPickerComponent._handleInput();
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.colorValueComponent.valueFormControl.value).toEqual('#g7g7g7');
+
+      testHostComponent.colorValueComponent.resetFormControl();
+
+      expect(testHostComponent.colorValueComponent.colorPickerComponent.value).toEqual('#ff3333');
+
+      expect(testHostComponent.colorValueComponent.form.valid).toBeFalsy();
+
+    });
+
+    it('should set a new display value', () => {
+
+      const newColor = new ReadColorValue();
+
+      newColor.color = '#d8d8d8';
+      newColor.id = 'updatedId';
+
+      testHostComponent.displayColorVal = newColor;
+
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent.colorValueComponent.colorPickerComponent.value).toEqual('#d8d8d8');
+
+      expect(testHostComponent.colorValueComponent.form.valid).toBeTruthy();
+
+    });
+
+    it('should unsubscribe when destroyed', () => {
+      expect(testHostComponent.colorValueComponent.valueChangesSubscription.closed).toBeFalsy();
+
+      testHostComponent.colorValueComponent.ngOnDestroy();
+
+      expect(testHostComponent.colorValueComponent.valueChangesSubscription.closed).toBeTruthy();
+    });
+
+  });
+
+  describe('create a color value', () => {
+
+    let testHostComponent: TestHostCreateValueComponent;
+    let testHostFixture: ComponentFixture<TestHostCreateValueComponent>;
+
+    let valueComponentDe: DebugElement;
+    let commentInputDebugElement: DebugElement;
+    let commentInputNativeElement;
+
+    beforeEach(() => {
+      testHostFixture = TestBed.createComponent(TestHostCreateValueComponent);
+      testHostComponent = testHostFixture.componentInstance;
+      testHostFixture.detectChanges();
+
+      expect(testHostComponent).toBeTruthy();
+      expect(testHostComponent.colorValueComponent).toBeTruthy();
+
+      const hostCompDe = testHostFixture.debugElement;
+
+      valueComponentDe = hostCompDe.query(By.directive(ColorValueComponent));
+      commentInputDebugElement = valueComponentDe.query(By.css('input.comment'));
+      commentInputNativeElement = commentInputDebugElement.nativeElement;
+    });
+
+    it('should create a value', () => {
+
+      expect(testHostComponent.colorValueComponent.colorPickerComponent.value).toEqual(null);
+
+      // simulate user input
+      const newColor = '#m5m5m5';
```

Comment:
this is not valid

Reply:
character "m" is not allowed.

Reply:
of course! just under my nose... too tired... thx